omod/src/test/java/org/openmrs/module/idgen/web/search/LogEntrySearchHandlerTest.java

```diff
@@ -0,0 +1,92 @@
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import org.junit.Assert;
+
+public class LogEntrySearchHandlerTest extends MainResourceControllerTest {
+    private IdentifierSourceService service;
+    public static String USER_UUID = "1010d442-e134-11de-babe-001e378eb67e";
+    public static String LOG_ENTRY_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a901";
+
+    @Before
+    public void setUp() throws Exception {
+        executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+        this.service = Context.getService(IdentifierSourceService.class);
+    }
+
+    @Override
+    public long getAllCount() {
+        return service.getLogEntries(null, null, null, null, null, null).size();
+    }
+
+    @Override
+    public String getURI() {
+        return "idgen/logentry";
+    }
+
+    @Override
+    public String getUuid() {
+        return "100892";
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryBySource() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("source", LOG_ENTRY_SOURCE_UUID);
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertNotNull(result);
+        Assert.assertEquals(3, Util.getResultsSize(result));
+
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByComment() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("comment", "New");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByIdentifier() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("identifier", "H9");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByDateGenerated() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("fromDate", "2016-10-01 12:00:00.433");
+        req.addParameter("toDate", "2017-09-30 12:00:00.433");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByGeneratedBy() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("generatedBy", USER_UUID);
```

Comment:
Change the dataset file in a way that not all the 5 rows have the same generated by. That is what will make this test more realistic. You can have like two or three rows with the generated by expected by this test.
Does it make sense?

Reply:
Yes it does, made the recommended changes